### ISSUE TITLE
FleatherField cursor does not stay visible when focused

### ISSUE THREAD

**Issue by kane-knowby** [fleather-editor/fleather#421]
When the cursor moves outside of the view (or behind the keyboard), the parent scrollable should scroll so that the cursor stays in view, the same way that TextFormField does.
 
## Steps to Reproduce

- Add a FleatherField to a ListView, with plenty of empty space above and below the field to scroll.
- Type in the FleatherField until the cursor disappears beneath the keyboard

## Environment

- OS: iOS/Web/Android
- Flutter version: 3.24
- Fleather version: 1.18.0

**Comment by amantoux** [comment]
@kane-knowby do you mind posting some code? Many thanks

**Comment by amantoux** [comment]
@kane-knowby I managed to reproduce
The thing here is that the scroll controller of the parent widget should not be controlled by the `FleatherField` as it depends on the position of the Field in the position of `FleatherField` with the parent view port.
You can try repositioning the parent scroll controller as the field grows

**Comment by amantoux** [comment]
Closing as stale for some time
Feel free to re-open

**Comment by Lomedhi** [comment]
Hi. I also want to give you master devs @Amir-P @amantoux some well deserved props because this package is amazing from where you took it and very sufficient for my use case. I also am having the same scrolling to the viewport issue with cursor going below the keyboard.  

I did some research into the matter and it seems a similar issue was reported on the original repo project [zefyr](https://github.com/memspace/zefyr/issues/485). 

From the thread, there is some proposed solution with considering the offset with and without the keyboard, but I believe this is for pack [flutter_quill](https://github.com/singerdmx/flutter-quill/pull/111). 

Here is the [implemented solution code](https://github.com/singerdmx/flutter-quill/commit/471c1cb7ee542977b777b0fc10cbaa5cfb35ddea).

I tried testing and debugging my own implementation (container,column, SingleChildScrollView,scaffold different parent child code, expanded and/or scrollable as true/false, set and not setting maxHeight), but got the same result of a non scrolling past the scroll container but also not scrolling the cursor above the keyboard. I can leave code snippet, but if you wrap fleatherfield in a listview and then wrap listview in a container is what the code would in essence be with other children in the listview as well.
